src/screens/CreateCommunity/index.js

```diff
@@ -92,12 +93,16 @@ const CreateCommunity = () => {
   }, []);
 
   // eslint-disable-next-line consistent-return
-  const next = () => {
+  const next = async () => {
     if (name && name.length >= MINIMUM_NAME_LENGTH && typeFetch === 'available') {
-      navigation.replace(NavigationConstants.CREATE_POST_SCREEN, {
-        isCreateCommunity: true,
-        topic: name
-      });
+      const response = await submitCommunityName(name);
+      if (response.success) {
+        navigation.replace('ContactScreen', {
+          isCreateCommunity: true,
+          topicCommunityId: response.topic_id,
+          topicCommunityName: name
+        });
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for navigating to the next screen after submitting the community name has been updated. It now checks if the response from `submitCommunityName` was successful before proceeding. This is a good practice as it ensures that the application doesn't proceed when there are errors. However, there seems to be no error handling if `submitCommunityName` fails. Consider adding an else clause to handle any failures from `submitCommunityName`.
```diff
  const next = async () => {
     if (name && name.length >= MINIMUM_NAME_LENGTH && typeFetch === 'available') {
       const response = await submitCommunityName(name);
       if (response.success) {
         navigation.replace('ContactScreen', {
           isCreateCommunity: true,
           topicCommunityId: response.topic_id,
           topicCommunityName: name
         });
+      } else {
+        // Handle failure case here
+      }
     } else {
       if (!name) {
         return showMessage({
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->